corehq/ex-submodules/casexml/apps/case/models.py

```diff
@@ -604,7 +606,19 @@ def apply_updates(self, update_action):
         Applies updates to a case
         """
         for k, v in update_action.updated_known_properties.items():
-            setattr(self, k, v)
+            try:
+                setattr(self, k, v)
+            except BadValueError as e:
+                soft_assert(notify_admins=True)(False, u'Error applying {}: {}'.format(
+                    unicode(update_action), e
+                ))
+                # todo: awkward hard-coded handling for legacy forms - can remove if these
+                # assertions go away: http://manage.dimagi.com/default.asp?182759
+                if k == 'opened_on' and isinstance(v, basestring):
+                    # this will reraise the exception if this wasn't a string date
+                    setattr(self, k, iso_string_to_datetime(v))
```

Comment:
it's actually [part of the spec](https://github.com/dimagi/commcare/wiki/casexml20) (as `date_opened`). it's not totally clear to me why in this particular instance it is failing, whereas in most instances it works properly. it might be that nobody uses it, or it might be something funny with this form (it's failing on a rebuild from a case update in 2012). either way, i figured the `soft_assert` might shed more light and if it comes up a bunch we can do something smarter